int64validator/at_most_sum_of_test.go

```diff
@@ -0,0 +1,144 @@
+		"not an Int64": {
+			val:         types.Bool{Value: true},
+			expectError: true,
+		},
+		"unknown Int64": {
+			val: types.Int64{Unknown: true},
+		},
+		"null Int64": {
+			val: types.Int64{Null: true},
+		},
+		"valid integer as Int64 more than sum of attributes": {
+			val: types.Int64{Value: 11},
+			attributesToSumPaths: []*tftypes.AttributePath{
+				tftypes.NewAttributePath().WithAttributeName("one"),
+				tftypes.NewAttributePath().WithAttributeName("two"),
+			},
+			requestConfigRaw: map[string]tftypes.Value{
+				"one": tftypes.NewValue(tftypes.Number, 5),
+				"two": tftypes.NewValue(tftypes.Number, 5),
+			},
+			expectError: true,
+		},
+		"valid integer as Int64 equal to sum of attributes": {
+			val: types.Int64{Value: 10},
+			attributesToSumPaths: []*tftypes.AttributePath{
+				tftypes.NewAttributePath().WithAttributeName("one"),
+				tftypes.NewAttributePath().WithAttributeName("two"),
+			},
+			requestConfigRaw: map[string]tftypes.Value{
+				"one": tftypes.NewValue(tftypes.Number, 5),
+				"two": tftypes.NewValue(tftypes.Number, 5),
+			},
+		},
+		"valid integer as Int64 less than sum of attributes": {
+			val: types.Int64{Value: 7},
+			attributesToSumPaths: []*tftypes.AttributePath{
+				tftypes.NewAttributePath().WithAttributeName("one"),
+				tftypes.NewAttributePath().WithAttributeName("two"),
+			},
+			requestConfigRaw: map[string]tftypes.Value{
+				"one": tftypes.NewValue(tftypes.Number, 4),
+				"two": tftypes.NewValue(tftypes.Number, 4),
+			},
+		},
+		"valid integer as Int64 less than sum of attributes, when one summed attribute is null": {
+			val: types.Int64{Value: 8},
+			attributesToSumPaths: []*tftypes.AttributePath{
+				tftypes.NewAttributePath().WithAttributeName("one"),
+				tftypes.NewAttributePath().WithAttributeName("two"),
+			},
+			requestConfigRaw: map[string]tftypes.Value{
+				"one": tftypes.NewValue(tftypes.Number, nil),
+				"two": tftypes.NewValue(tftypes.Number, 9),
+			},
+		},
+		"valid integer as Int64 does not return error when all attributes are null": {
+			val: types.Int64{Null: true},
+			attributesToSumPaths: []*tftypes.AttributePath{
+				tftypes.NewAttributePath().WithAttributeName("one"),
+				tftypes.NewAttributePath().WithAttributeName("two"),
+			},
+			requestConfigRaw: map[string]tftypes.Value{
+				"one": tftypes.NewValue(tftypes.Number, nil),
+				"two": tftypes.NewValue(tftypes.Number, nil),
+			},
+		},
+		"valid integer as Int64 less than sum of attributes, when one summed attribute is unknown": {
+			val: types.Int64{Value: 8},
+			attributesToSumPaths: []*tftypes.AttributePath{
+				tftypes.NewAttributePath().WithAttributeName("one"),
+				tftypes.NewAttributePath().WithAttributeName("two"),
+			},
+			requestConfigRaw: map[string]tftypes.Value{
+				"one": tftypes.NewValue(tftypes.Number, tftypes.UnknownValue),
+				"two": tftypes.NewValue(tftypes.Number, 9),
+			},
+		},
+		"valid integer as Int64 does not return error when all attributes are unknown": {
```

Comment:
This test case has me wondering what happens if the `val` is `types.Int64{Value: 1}` and all attributes are unknown? I believe the current code would return an error that it is greater than `int64`'s zero value of `0`. I would expect the validator to elide an error in that case, because it cannot really determine whether it is valid or not.

Aside: Admittedly weird usage because I guess the number would have to be negative, but I think AtLeastSumOf has the inverse of the same potential issue.